apps/pages/models.py

```diff
@@ -0,0 +1,176 @@
+from django.apps import apps
+from django.db import models
+from django.utils import timezone
+
+from apps.common.models import IndexedTimeStampedModel
+
+
+class AbstractPage(IndexedTimeStampedModel):
+    name = models.CharField(max_length=255)
+    title = models.CharField(max_length=255)
+
+    header_bg_image = models.ImageField(null=True, blank=True)
+    header_logo = models.ImageField(null=True, blank=True)
+    header_link = models.URLField(null=True, blank=True)
+
+    styles = models.ForeignKey("pages.Style", null=True, blank=True, on_delete=models.SET_NULL)
+
+    elements = models.JSONField(null=True, blank=True)
+
+    show_benefits = models.BooleanField(default=False)
+
+    # lookup with donor_benefit.donationpages / donor_benefit.templates
+    donor_benefits = models.ForeignKey(
+        "pages.DonorBenefit",
+        null=True,
+        blank=True,
+        related_name="%(class)ss",
```

Comment:
This would be a more contemporary way of doing the string formatting here (and to my mind, more readable):

```suggestion
        related_name=f"{class}s",
```

Reply:
That won't actually work here. It expects a `class` variable to be defined, whereas %(class)s seems to be lazy.